gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/web/internal/configureWebApplication.kt

```diff
@@ -23,13 +23,34 @@ internal fun Project.configureWeb(
     composeExt: ComposeExtension,
 ) {
     val webExt = composeExt.extensions.getByType(WebExtension::class.java)
+
+    // here we check all dependencies (including transitive)
+    // If there is compose.ui, then skiko is required!
+    val shouldRunUnpackSkiko = project.provider {
+        var dependsOnComposeUi = false
+        project.configurations.matching {
+            val isWasmOrJs = it.name.contains("js", true) ||
+                    it.name.contains("wasm", true)
+
+            it.isCanBeResolved && isWasmOrJs
+        }.all {
+            val match = it.incoming.artifacts.resolvedArtifacts.get().any {
+                it.id.componentIdentifier.toString().contains("org.jetbrains.compose.ui")
+            }
+
+            dependsOnComposeUi = dependsOnComposeUi || match
```

Comment:
I'd ask use explicit parameters instead of 'it'

Reply:
done